### DISCUSSION TITLE
Preview not works in production

### DISCUSSION THREAD

**Discussion by qkreltms** [vercel/next.js#18521] [Help]
I'm trying to implement preview mode, it works well in development, but not in production which run on http://localhost/3000. It seems cookie doesn't set in production. is it bug?

What I've done is ```yarn build```, after building app I ran ```yarn start```

Implementation: 
https://github.com/qkreltms/nextjs-tutorial02

https://codesandbox.io/s/priceless-butterfly-8jkim?file=/pages/index.js

Let me know if you need more info.

**Comment by qkreltms** [comment]
Seems bug for Chrome, because when I ran it on Firefox, it works perfectly. I'am wondering it occurs only in localhost.

ps: I checked it other browsers like IE, Edge, It's not works

**Reply by vasco3** [reply]
works for me in localhost but not when deployed in Vercel (safari)

**Reply by leerob** [reply]
Here's a completed example showing preview mode: https://github.com/vercel/preview-mode-demo

Hope this helps!

**Comment by justintemps** [comment]
I'm having the exact same issue. It seems like `getStaticProps` is not called reliably in production when preview mode cookies are set.

**Localhost/Development**
Everything works fine, es expected. 

- Requests to localhost:3000/api/preview?token={myToken}&documentId={myDocumentId} redirects appropriately with preview mode cookies set. getStaticProps is called

**Localhost/Production**

Preview mode works fine in Firefox, but not Chrome. In both browsers, I can see that `__next_preview_data` and `__prerender_bypass` cookies have been set.

**On Server/Production**
Preview mode doesn't work in any browser even though I can see that  `__next_preview_data` and `__prerender_bypass` cookies have been set.

Really confused here....

**Reply by justintemps** [reply]
The strange thing is that this is only happening for dynamic routes. Preview mode in static routes works fine, but in `stories/[uid].tsx`, a page with `getStaticPaths`, preview mode cookies do not cause getStaticProps to be called.

**Reply by justintemps** [reply]
Opened a ticket here as I think it could be a bug: https://github.com/vercel/next.js/issues/19714
